examples/default.yaml

```diff
@@ -199,6 +199,19 @@ containerd:
 # - mode: boot
 #   script: |
 #     systemctl disable NetworkManager-wait-online.service
+# # `dependency` is executed before the regular dependency resolution workflow in
+# # pkg/cidata/cidata.TEMPLATE.d/boot/30-install-packages.sh
+# - mode: dependency
+#   script: |
+#     #!/bin/bash
+#     dnf config-manager --add-repo ...
+#     dnf install ...
+
+# `skipDefaultDependencyResolution` can be used with `dependency` mode provisioning scripts to skip
+# the regular dependency resolution workflow in pkg/cidata/cidata.TEMPLATE.d/boot/30-install-packages.sh
+# and provide a fully customizable replacement
+# 🟢 Builtin default: false
+# skipDefaultDependencyResolution: true
```

Comment:
This is quite a mouthful, and also feels somewhat disconnected here. I'm wondering if this would make more sense to make this an option on the `dependency` provisioning script.

I don't have strong opinions, it just feels a bit clunky to have this weirdly-named option at the top level, that only makes sense when you also have some other bits of configuration defined.

Reply:
Good point, addressed in latest revision. I added a new field to the Provisioning scripts in general, but invalidate it when used on non-dependency scripts.